cmake/project_version.cmake

```diff
@@ -0,0 +1,28 @@
+include_guard()
+
+function(obtain_project_version resultVarVersion resultVarGitHash)
+    if (GIT_FOUND)
+        execute_process(
+                COMMAND ${GIT_EXECUTABLE} --git-dir=${CMAKE_CURRENT_SOURCE_DIR}/.git describe --abbrev=0 --tags
+                OUTPUT_VARIABLE VERSION_STRING
+                OUTPUT_STRIP_TRAILING_WHITESPACE
+        )
+        execute_process(
+                COMMAND ${GIT_EXECUTABLE} rev-parse HEAD
+                RESULT_VARIABLE git_result
+                OUTPUT_VARIABLE GIT_HASH
+                OUTPUT_STRIP_TRAILING_WHITESPACE
+        )
+        if (NOT git_result)
+            set(${resultVarGitHash} ${GIT_HASH} PARENT_SCOPE)
+            message(STATUS "Building git hash: ${GIT_HASH}")
+        endif()
+    endif ()
+
+    if (NOT VERSION_STRING) # Non DEV build relays on VERSION file updated by release CI
+        file(READ "${CMAKE_CURRENT_SOURCE_DIR}/VERSION" VERSION_STRING)
+    endif ()
```

Comment:
In non dev builds, those with no git, the version will be read from a VERSION file in the root of the SDK sourcetree. This file is expected to be updated by CI at the release process. The placement in the root is to have easy access to it for human readers.